velox/exec/Window.cpp

```diff
@@ -99,6 +87,152 @@ Window::Window(
   createWindowFunctions(windowNode, inputType);
 }
 
+vector_size_t findKOffset(
+    vector_size_t& currentRow,
+    std::optional<FrameArg>& frameArg,
+    std::optional<vector_size_t>& index,
+    bool addOffset) {
+  if (!frameArg || !frameArg->constantValue) {
```

Comment:
These check's shouldn't be done each time in the function.

Reply:
Thanks for pointing it out, moved these checks to createWindowFunctions. However, I want to remove these checks since it is always set based on the frame type, would that be fine?